lib/game_setting_setter.rb

```diff
@@ -0,0 +1,33 @@
+require_relative './game_presenter'
+require_relative './game_setting'
+require_relative './messages'
+
+class GameSettingSetter
+    include Messages
+    attr_accessor :settings, :setting_types, :game_presenter
+
+    INITIAL_SETTING_TYPES = [
+            GameSetting.new(setting: BoardSize.new()),
+            GameSetting.new(setting: OpponentType.new()),
+    ]
+
+    def initialize(
+        setting_types: INITIAL_SETTING_TYPES,
+        game_presenter: ConsoleIO.new
+    )
+        @setting_types = setting_types
+        @game_presenter = game_presenter
+        @settings = gets
```

Comment:
It's a bit strange that we are calling `gets` inside this constructor as it has side effects. We should only be initializing a class inside a constructor and leave any actual operations as methods.

Reply:
Agreed